### PR TITLE
Make it API compatible with the existing vga crate.

### PR DESCRIPTION
**Pull request by steveklabnik**
Changes:
1. Adding the kprintln!/kprint! macros.
2. Adding a spinlock to hold a global, so that those work.

This also then requires `const fn` and so makes things nightly-only. For
now!

r? @ashleygwilliams

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -80,6 +88,44 @@ impl Write for Vga {
     }
 }
 
+pub static BUFFER: Mutex<Vga> = Mutex::new(
+    Vga {
+        location: 0xb8000 as *mut u8,
+        buffer: [0; ROWS * COLS * 2],
```
Comment (ashleygwilliams) [inline src/lib.rs]: update to use `COL_BYTES`

File: src/lib.rs [hunk 2]
```diff
@@ -80,6 +88,44 @@ impl Write for Vga {
...
+
+/// Prints something to the screen, with a trailing newline.
+///
+/// # Examples
+///
+/// ```
+/// kprintln!("Hello, world!");
+/// ```
+#[macro_export]
+macro_rules! kprintln {
```
Comment (ashleygwilliams) [inline src/lib.rs]: can we put this in a separate `util.rs`

File: src/lib.rs [hunk 3]
```diff
@@ -80,6 +88,44 @@ impl Write for Vga {
...
+}
+
+/// Prints something to the screen.
+///
+/// # Examples
+///
+/// ```
+/// kprint!("Hello, world!");
+/// ```
+#[macro_export]
```
Comment (ashleygwilliams) [inline src/lib.rs]: also this

**Review by ashleygwilliams — CHANGES_REQUESTED** [review]
some nits:
- move the stuff that has nothing to do with the lib out of `lib.rs`
- update to use new `COL_BYTES` constant

File: src/lib.rs (hunk 2 above)
Comment (ashleygwilliams) [reply src/lib.rs]: or `print.rs`

Comment (steveklabnik) [reply src/lib.rs]: 1. `util.rs` is a bad name, but I'm not sure if you're married to the idea of that name or it's just to not have it in this file
2. I actually want to kill this entirely, but that requires upstream refactoring first.

so, my vote would be to just leave it in for now.

Comment (ashleygwilliams) [reply src/lib.rs]: name it whatever you want ✨ but i want it out of this file

Comment (ashleygwilliams) [reply src/lib.rs]: OR: spend 5min trying to figure out how to get rid of it, and then if we can at least put it all the way at the end

**Comment by ashleygwilliams** [comment]
🎉

**Review by ashleygwilliams — APPROVED** [review]

**Pull request merged**